src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -315,12 +332,136 @@ private static void printLocales() {
+                printAllSecurityConfig();
+            }
+
+        }
+    }
+
+    // A non-verbose description of some core security configuration settings
+    private static void printSecuritySummarySettings() {
+        ostream.println("Security settings summary: " + "\n" +
+                INDENT + "Use \"-XshowSettings:security\" for verbose details\n");
+        printSecurityProviderConfig(false);
+        printSecurityTLSConfig(false);
+    }
+
+    private static void printAllSecurityConfig() {
+        ostream.println("Security settings:\n");
+        printSecurityProperties();
+        printSecurityProviderConfig(true);
+        printSecurityTLSConfig(true);
+    }
+
+    private static void printSecurityProperties() {
+        ostream.println(INDENT + "Security properties:");
+        Properties p = SharedSecrets.getJavaSecurityPropertiesAccess().getInitialProperties();
+        for (String key : p.stringPropertyNames().stream().sorted().toList()) {
+            String val = p.getProperty(key);
+            if (val.contains(",") && val.length() > 60) {
+                // split lines longer than 60 chars which have multiple values
+                ostream.println(TWOINDENT + key + "=");
+                List.of(val.split(",")).forEach(
+                        s -> ostream.println(THREEINDENT + s.trim() + ","));
+            } else {
+                ostream.println(TWOINDENT + key + "=" + val);
+            }
+        }
+        ostream.println();
+    }
+
+    private static void printSecurityTLSConfig(boolean verbose) {
+        SSLSocket ssls;
+        try {
+            ssls = (SSLSocket)
+                    SSLContext.getDefault().getSocketFactory().createSocket();
+        } catch (IOException | NoSuchAlgorithmException e) {
+            throw new InternalError("Failed to create SSL socket");
+        }
+
+        ostream.println(INDENT + "Security TLS configuration:");
+        ostream.println(TWOINDENT + "Enabled Protocols:");
+        for (String s : ssls.getEnabledProtocols()) {
+            System.out.println(THREEINDENT + s);
+        }
+
+        if (verbose) {
+            System.out.println("\n" + TWOINDENT + "Enabled Cipher Suites:");
+            for (String s : ssls.getEnabledCipherSuites()) {
+                System.out.println(THREEINDENT + s);
+            }
+        }
+        ostream.println();
+    }
+
+    private static void printSecurityProviderConfig(boolean verbose) {
+        ostream.println(INDENT + "Security provider static configuration:");
+        for (Provider p : Security.getProviders()) {
+            if (verbose) {
+                // separate the views out
+                ostream.println(TWOINDENT + "-".repeat(40));
+            }
+            ostream.println(TWOINDENT + "Provider name: " + p.getName());
+            if (verbose) {
+                ostream.println(TWOINDENT + PROV_INFO_STRING + wrappedString(p.getInfo(), 80));
```

Comment:
Wrapping the args at 80 might still produce line that is quite long.
Perhaps wrap the string after it was concatenated.

Reply:
logic updated. I think that's what you were hinting at Roger.